### PR TITLE
Add user permissions subscription

### PR DESCRIPTION
**Pull request by rochaeb**
# Problem
referenced issue(s):

# Solution

# Testing

### CODE REVIEW

**Review by rssk — CHANGES_REQUESTED** [review]

File: packages/coinstac-ui/app/render/components/dashboard/dashboard.jsx [hunk 1]
```diff
@@ -407,6 +407,14 @@ class Dashboard extends Component {
     }
   }
 
+  componentDidUpdate(prevProps) {
+    if (this.props.currentUser &&
+      (!prevProps.currentUser || prevProps.currentUser.permissions !== this.props.currentUser.permissions)
```
[the remark this answers is not in the document]
Comment (rochaeb) [reply packages/coinstac-ui/app/render/components/dashboard/dashboard.jsx]: Yes, I'm indeed only comparing references and this equality comparison probably has no utility... I was thinking of using Immutable to improve efficiency and making changes on the store only when necessary, but maybe we'll want this performance improvement only when we actually need it? What do you think?

File: packages/coinstac-ui/app/render/state/graphql/props.js [hunk 2]
```diff
@@ -169,10 +169,7 @@ export const userRolesProp = (name) => {
         variables: {
           userId, table, doc, role,
         },
-      })
-        .then(({ data: { [name]: { permissions } } }) => {
```
[the remark this answers is not in the document]
Comment (rochaeb) [reply packages/coinstac-ui/app/render/state/graphql/props.js:173]: What was happening here is that when a user created a consortium the api was processing the permission changes for that user and returning this. With this a permission change on the store was triggered. As we now have the subscription for this, I removed the this code.

I guess I could also remove the permissions "calculation" in the api mutation for adding/updating user roles.

File: packages/coinstac-ui/app/render/components/dashboard/dashboard.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (rochaeb) [reply packages/coinstac-ui/app/render/components/dashboard/dashboard.jsx:410]: This indeed is comparing references. As this is on the componentDidUpdate lifecycle method, this check prevents updating the store unnecessarily when the currentUser hasn't changed. We could improve this later (performance-wise) using Immutable for checking if the permissions did change when the user metadata subscription is triggered, but maybe we'll want to worry with this if we actually feel we need a performance improvement on this.

[the remark this answers is not in the document]
Comment (rssk) [reply packages/coinstac-ui/app/render/components/dashboard/dashboard.jsx:410]: 👍

**Review by rssk — APPROVED** [review]

**Pull request merged**